src/Dhcp.cpp

```diff
@@ -5,9 +5,13 @@
 #include "Ethernet.h"
 #include "Dhcp.h"
 #include "utility/w5100.h"
+#ifndef STM32BUILD
 #include <avr/wdt.h>
+#else
+#include <IWatchdog.h>
+#endif
 
-int DhcpClass::beginWithDHCP(uint8_t *mac, const char *hostname = nullptr, unsigned long timeout = 60000, unsigned long responseTimeout = 4000, ProcessEventsCallback eventsCallback = nullptr)
+int DhcpClass::beginWithDHCP(uint8_t *mac, const char *hostname, unsigned long timeout, unsigned long responseTimeout, ProcessEventsCallback eventsCallback)
```

Comment:
@MilesCircuitworx Do you know why the defaults have been removed here?

Reply:
It seems to avoid a compilation error, which I think is because default arguments are only permitted at the function declaration rather than in the definition.

I've just tried building the firmware with the default arguments put back in and I get this error:
(it builds fine without)

```
/workspaces/cogs-box-firmware/arduino-cli-files/user/libraries/Ethernet/src/Ethernet.cpp:29:5: error: default argument given for parameter 2 of 'static int EthernetClass::begin(uint8_t*, const char*, long unsigned int, long unsigned int, ProcessEventsCallback)' [-fpermissive]
   29 | int EthernetClass::begin(uint8_t *mac, const char *hostname = nullptr, unsigned long timeout = 60000, unsigned long responseTimeout = 4000, ProcessEventsCallback eventsCallback = nullptr){
      |     ^~~~~~~~~~~~~
In file included from /workspaces/cogs-box-firmware/arduino-cli-files/user/libraries/Ethernet/src/Ethernet.cpp:22:
/workspaces/cogs-box-firmware/arduino-cli-files/user/libraries/Ethernet/src/Ethernet.h:92:14: note: previous specification in 'static int EthernetClass::begin(uint8_t*, const char*, long unsigned int, long unsigned int, ProcessEventsCallback)' here
   92 |   static int begin(uint8_t *mac, const char *hostname = nullptr, unsigned long timeout = 60000, unsigned long responseTimeout = 4000, ProcessEventsCallback eventsCallback = nullptr);
      |              ^~~~~
/workspaces/cogs-box-firmware/arduino-cli-files/user/libraries/Ethernet/src/Ethernet.cpp:29:5: error: default argument given for parameter 3 of 'static int EthernetClass::begin(uint8_t*, const char*, long unsigned int, long unsigned int, ProcessEventsCallback)' [-fpermissive]
   29 | int EthernetClass::begin(uint8_t *mac, const char *hostname = nullptr, unsigned long timeout = 60000, unsigned long responseTimeout = 4000, ProcessEventsCallback eventsCallback = nullptr){
      |     ^~~~~~~~~~~~~
/workspaces/cogs-box-firmware/arduino-cli-files/user/libraries/Ethernet/src/Ethernet.h:92:14: note: previous specification in 'static int EthernetClass::begin(uint8_t*, const char*, long unsigned int, long unsigned int, ProcessEventsCallback)' here
   92 |   static int begin(uint8_t *mac, const char *hostname = nullptr, unsigned long timeout = 60000, unsigned long responseTimeout = 4000, ProcessEventsCallback eventsCallback = nullptr);
      |              ^~~~~
/workspaces/cogs-box-firmware/arduino-cli-files/user/libraries/Ethernet/src/Ethernet.cpp:29:5: error: default argument given for parameter 4 of 'static int EthernetClass::begin(uint8_t*, const char*, long unsigned int, long unsigned int, ProcessEventsCallback)' [-fpermissive]
   29 | int EthernetClass::begin(uint8_t *mac, const char *hostname = nullptr, unsigned long timeout = 60000, unsigned long responseTimeout = 4000, ProcessEventsCallback eventsCallback = nullptr){
      |     ^~~~~~~~~~~~~
/workspaces/cogs-box-firmware/arduino-cli-files/user/libraries/Ethernet/src/Ethernet.h:92:14: note: previous specification in 'static int EthernetClass::begin(uint8_t*, const char*, long unsigned int, long unsigned int, ProcessEventsCallback)' here
   92 |   static int begin(uint8_t *mac, const char *hostname = nullptr, unsigned long timeout = 60000, unsigned long responseTimeout = 4000, ProcessEventsCallback eventsCallback = nullptr);
      |              ^~~~~
/workspaces/cogs-box-firmware/arduino-cli-files/user/libraries/Ethernet/src/Ethernet.cpp:29:5: error: default argument given for parameter 5 of 'static int EthernetClass::begin(uint8_t*, const char*, long unsigned int, long unsigned int, ProcessEventsCallback)' [-fpermissive]
   29 | int EthernetClass::begin(uint8_t *mac, const char *hostname = nullptr, unsigned long timeout = 60000, unsigned long responseTimeout = 4000, ProcessEventsCallback eventsCallback = nullptr){
      |     ^~~~~~~~~~~~~
/workspaces/cogs-box-firmware/arduino-cli-files/user/libraries/Ethernet/src/Ethernet.h:92:14: note: previous specification in 'static int EthernetClass::begin(uint8_t*, const char*, long unsigned int, long unsigned int, ProcessEventsCallback)' here
   92 |   static int begin(uint8_t *mac, const char *hostname = nullptr, unsigned long timeout = 60000, unsigned long responseTimeout = 4000, ProcessEventsCallback eventsCallback = nullptr);
      |              ^~~~~
/workspaces/cogs-box-firmware/arduino-cli-files/user/libraries/Ethernet/src/Dhcp.cpp:14:5: error: default argument given for parameter 2 of 'int DhcpClass::beginWithDHCP(uint8_t*, const char*, long unsigned int, long unsigned int, ProcessEventsCallback)' [-fpermissive]
   14 | int DhcpClass::beginWithDHCP(uint8_t *mac, const char *hostname = nullptr, unsigned long timeout = 60000, unsigned long responseTimeout = 4000, ProcessEventsCallback eventsCallback = nullptr){
      |     ^~~~~~~~~
In file included from /workspaces/cogs-box-firmware/arduino-cli-files/user/libraries/Ethernet/src/Dhcp.cpp:5:
/workspaces/cogs-box-firmware/arduino-cli-files/user/libraries/Ethernet/src/Ethernet.h:329:7: note: previous specification in 'int DhcpClass::beginWithDHCP(uint8_t*, const char*, long unsigned int, long unsigned int, ProcessEventsCallback)' here
  329 |   int beginWithDHCP(uint8_t *, const char * = nullptr, unsigned long timeout = 60000, unsigned long responseTimeout = 4000, ProcessEventsCallback eventsCallback = nullptr);
      |       ^~~~~~~~~~~~~
/workspaces/cogs-box-firmware/arduino-cli-files/user/libraries/Ethernet/src/Dhcp.cpp:14:5: error: default argument given for parameter 3 of 'int DhcpClass::beginWithDHCP(uint8_t*, const char*, long unsigned int, long unsigned int, ProcessEventsCallback)'
   14 | int DhcpClass::beginWithDHCP(uint8_t *mac, const char *hostname = nullptr, unsigned long timeout = 60000, unsigned long responseTimeout = 4000, ProcessEventsCallback eventsCallback = nullptr){
      |     ^~~~~~~~~
/workspaces/cogs-box-firmware/arduino-cli-files/user/libraries/Ethernet/src/Ethernet.h:329:7: note: previous specification in 'int DhcpClass::beginWithDHCP(uint8_t*, const char*, long unsigned int, long unsigned int, ProcessEventsCallback)' here
  329 |   int beginWithDHCP(uint8_t *, const char * = nullptr, unsigned long timeout = 60000, unsigned long responseTimeout = 4000, ProcessEventsCallback eventsCallback = nullptr);
      |       ^~~~~~~~~~~~~
/workspaces/cogs-box-firmware/arduino-cli-files/user/libraries/Ethernet/src/Dhcp.cpp:14:5: error: default argument given for parameter 4 of 'int DhcpClass::beginWithDHCP(uint8_t*, const char*, long unsigned int, long unsigned int, ProcessEventsCallback)'
   14 | int DhcpClass::beginWithDHCP(uint8_t *mac, const char *hostname = nullptr, unsigned long timeout = 60000, unsigned long responseTimeout = 4000, ProcessEventsCallback eventsCallback = nullptr){
      |     ^~~~~~~~~
/workspaces/cogs-box-firmware/arduino-cli-files/user/libraries/Ethernet/src/Ethernet.h:329:7: note: previous specification in 'int DhcpClass::beginWithDHCP(uint8_t*, const char*, long unsigned int, long unsigned int, ProcessEventsCallback)' here
  329 |   int beginWithDHCP(uint8_t *, const char * = nullptr, unsigned long timeout = 60000, unsigned long responseTimeout = 4000, ProcessEventsCallback eventsCallback = nullptr);
      |       ^~~~~~~~~~~~~
/workspaces/cogs-box-firmware/arduino-cli-files/user/libraries/Ethernet/src/Dhcp.cpp:14:5: error: default argument given for parameter 5 of 'int DhcpClass::beginWithDHCP(uint8_t*, const char*, long unsigned int, long unsigned int, ProcessEventsCallback)' [-fpermissive]
   14 | int DhcpClass::beginWithDHCP(uint8_t *mac, const char *hostname = nullptr, unsigned long timeout = 60000, unsigned long responseTimeout = 4000, ProcessEventsCallback eventsCallback = nullptr){
      |     ^~~~~~~~~
/workspaces/cogs-box-firmware/arduino-cli-files/user/libraries/Ethernet/src/Ethernet.h:329:7: note: previous specification in 'int DhcpClass::beginWithDHCP(uint8_t*, const char*, long unsigned int, long unsigned int, ProcessEventsCallback)' here
  329 |   int beginWithDHCP(uint8_t *, const char * = nullptr, unsigned long timeout = 60000, unsigned long responseTimeout = 4000, ProcessEventsCallback eventsCallback = nullptr);
      |       ^~~~~~~~~~~~~
```